heal.js

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env node
+
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+'use strict';
+
+var ClusterManager = require('./lib/cluster.js');
+var assertNoError = require('./lib/util.js').assertNoError;
+var program = require('commander');
+
+function safeParseInt(number, defaultValue) {
+    number = parseInt(number, 10);
+    if (isNaN(number)) {
+        return defaultValue;
+    } else {
+        return number;
+    }
+}
+
+function main() {
+    program
+        .description('Start a partition heal coordinated by the target node')
+        .option('--tchannel-v1')
+        .option('--quiet')
+        .option('--tries <tries>', 'Number of times to try the heal', safeParseInt, 10)
```

Comment:
There is a benefit of having a lower interval and higher number of retries, on average the command will finish sooner with partitions still being healed on a bigger cluster.

So only if there is a very heavy load added on calling the heal every second for 10 seconds I'd rather keep it like this. There are little services that run 1k nodes.